addon/services/observer-manager.js

```diff
@@ -0,0 +1,31 @@
+import Service from '@ember/service';
+import IntersectionObserverAdmin from 'intersection-observer-admin';
+
+export default class ObserverManagerService extends Service {
+  _admin;
+
+  constructor() {
+    super(...arguments);
+    this._admin = new IntersectionObserverAdmin();
+  }
```

Comment:
**Style nits** (by definition not blocking 😅):

1. For this kind of thing I think it's clearer (and it's definitely *briefer*!) to just assign the class field directly.
2. We can use true privacy here as long as we include the correct Babel plugins. (If the addon is using a sufficiently recent version of ember-cli-babel, that's done by default.)

If you take those suggestions, you end up with this:

```suggestion
  #admin = new IntersectionObserverAdmin();
```

I note, however, that you're likely doing it this way for the sake of testing, in which case you can ignore this approach. (There's a weird corner here with services today, in that you really can't parameterize them at all, so normal approaches where you handle it via argument to constructor with a default don't work!)

Reply:
yea, we're stubbing methods on the admin for testing so I think the current approach is probably fine. It's cool that you can do that, though!

Reply:
Yeah, given which I'd probably still go for the shorthand, just with `_` instead:

```suggestion
  _admin = new IntersectionObserverAdmin();
```